src/flake8_tidy_imports/__init__.py

```diff
@@ -63,7 +67,18 @@ def parse_options(cls, options: Any) -> None:
             module, message = line.split("=", 1)
             module = module.strip()
             message = message.strip()
-            cls.banned_modules[module] = message
+            # Entries that end with .* will be treated as prefixes for matching
+            # Store them in a list first, so we can sort them before adding to the dict
+            if module.endswith(".*"):
+                prefix = module[:-2]
+                cls.banned_modules[prefix] = message
+                banned_patterns.append((module, message))
+            else:
+                cls.banned_modules[module] = message
+
+        banned_patterns.sort(key=lambda x: len(x[0]), reverse=True)
```

Comment:
This is so that if we have e.g. `foo.*` and `foo.bar.*` banned and try `import foo.bar.baz`, the one that matches with higher priority is `foo.bar.*`